### PR TITLE
add stats for perennial

### PR DESCRIPTION
**Pull request by wangyangjun**
**Changes**:
- add stats data for perennial

### CODE REVIEW

File: src/scripts/stats.py [hunk 1]
```diff
@@ -150,20 +176,26 @@ async def main(config_key):
...
-    df_trader['dateTs'] = df_trader['timestamp'].apply(lambda x: int(int(x) / 86400) * 86400)
-    df_trader['cumulativeTraders'] = (~df_trader['accountId' if config_key == 'v3' else 'account'].duplicated()).cumsum()
-    df_trader_agg = df_trader.groupby('dateTs')['accountId' if config_key == 'v3' else 'account'].nunique().reset_index()
-    df_trader_agg.columns = ['timestamp', 'uniqueTraders']
-    df_trader_agg['cumulativeTraders'] = df_trader.groupby('dateTs')['cumulativeTraders'].max().reset_index()['cumulativeTraders']
+    if config_key == 'perennial':
+        df_agg['volume'] = df_agg['longNotional'].astype(float) / 1000000 + df_agg['shortNotional'].astype(float) / 1000000
+        df_agg['uniqueTraders'] = df_agg['traders'].astype(int)
+        df_agg['cumulativeTraders'] = df_agg['uniqueTraders'].cumsum()
+        df_write = df_agg[['timestamp', 'volume', 'trades', 'cumulativeTrades', 'uniqueTraders', 'cumulativeTraders']]
```
Comment (LeifuChen) [inline src/scripts/stats.py:183]: We need to perform a `groupby` operation based on the `timestamp` column for the `df_agg` DataFrame.

File: src/scripts/stats.py [hunk 2]
```diff
@@ -150,20 +176,26 @@ async def main(config_key):
...
-    # Trader data query and processing
-    trader_query = config['queries']['traders']
-    df_trader = pd.DataFrame(await run_recursive_query(trader_query['query'], {'last_id': ''}, trader_query['accessor'], config['subgraph_endpoint'])).drop('id', axis=1).sort_values('timestamp')
-    df_trader['dateTs'] = df_trader['timestamp'].apply(lambda x: int(int(x) / 86400) * 86400)
-    df_trader['cumulativeTraders'] = (~df_trader['accountId' if config_key == 'v3' else 'account'].duplicated()).cumsum()
-    df_trader_agg = df_trader.groupby('dateTs')['accountId' if config_key == 'v3' else 'account'].nunique().reset_index()
-    df_trader_agg.columns = ['timestamp', 'uniqueTraders']
-    df_trader_agg['cumulativeTraders'] = df_trader.groupby('dateTs')['cumulativeTraders'].max().reset_index()['cumulativeTraders']
+    if config_key == 'perennial':
+        df_agg['volume'] = df_agg['longNotional'].astype(float) / 1000000 + df_agg['shortNotional'].astype(float) / 1000000
```
Comment (LeifuChen) [inline src/scripts/stats.py:180]: We might use `_` as a thousands separator in `1_000_000` for better readability.

File: src/scripts/stats.py (hunk 1 above)
Comment (wangyangjun) [reply src/scripts/stats.py:184]: good catch

Comment (LeifuChen) [reply src/scripts/stats.py:184]: Thanks for the changes. I am checking the fields in the `MarketAccumulation`. It seems that we only need to use the data filtered by `bucket: daily`. The duplicated timestamps come from the different bucket types.

File: src/scripts/stats.py [hunk 3]
```diff
@@ -148,22 +175,30 @@ async def main(config_key):
...
-    # Trader data query and processing
-    trader_query = config['queries']['traders']
-    df_trader = pd.DataFrame(await run_recursive_query(trader_query['query'], {'last_id': ''}, trader_query['accessor'], config['subgraph_endpoint'])).drop('id', axis=1).sort_values('timestamp')
-    df_trader['dateTs'] = df_trader['timestamp'].apply(lambda x: int(int(x) / 86400) * 86400)
-    df_trader['cumulativeTraders'] = (~df_trader['accountId' if config_key == 'v3' else 'account'].duplicated()).cumsum()
-    df_trader_agg = df_trader.groupby('dateTs')['accountId' if config_key == 'v3' else 'account'].nunique().reset_index()
-    df_trader_agg.columns = ['timestamp', 'uniqueTraders']
-    df_trader_agg['cumulativeTraders'] = df_trader.groupby('dateTs')['cumulativeTraders'].max().reset_index()['cumulativeTraders']
+    if config_key == 'perennial':
+        df_agg = df_agg.groupby('timestamp').sum().reset_index()
```
Comment (LeifuChen) [inline src/scripts/stats.py:180]: Have you checked the export results? The total volume seems off after this group operation. We might perform the unit conversion of longNotional and shortNotional before the aggregation.

**Review by LeifuChen — APPROVED** [review]
Thanks for the quick fix. Looks good to me.

**Pull request merged**